datadog_checks_base/datadog_checks/base/utils/discovery/cache.py

```diff
@@ -0,0 +1,27 @@
+# (C) Datadog, Inc. 2019-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import time
+
+
+class Cache:
+    def __init__(self, get_items_func, interval):
```

Comment:
What do you think of adding typing (even if it's not mandatory in the base check)?

Reply:
I tried to follow the line of code that we already have in the base check

Reply:
Looks like you're having issues with py2 because of that 🙁 If you want to keep them, you'll have to specify them as comments, e.g. https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/checks/win/winpdh_base.py#L63-L66

Reply:
Removed

Reply:
Yes, we have support for mypy typing but not native typing as that is a py3 feature